chapters/VK_EXT_present_timing/PresentTimeInfo.adoc

```diff
@@ -0,0 +1,184 @@
+  * pname:pTimingInfos is `NULL` or a pointer to an array of
+    sname:VkPresentTimingInfoEXT elements with pname:swapchainCount entries.
+    If not `NULL`, each element of pname:pTimingInfos contains timing
+    information for the presentation of the image corresponding to the entry
+    in the sname:VkPresentInfoKHR::pname:pImageIndices array.
+
+The semantics of target presentation time or duration vary depending on the presentation mode:
+
+  * ename:VK_PRESENT_MODE_FIFO_KHR.
+    Tearing cannot: be observed.
+    The first queued image is dequeued and presented if
+    both: 1) its associated wait semaphore(s) have signaled, and 2)
+    its target present time is less-than or equal-to the current time.
+  * ename:VK_PRESENT_MODE_FIFO_RELAXED_KHR.
+    For images that are presented in time to be displayed at the next
+    vertical blanking period, the semantics are identical as for
+    ename:VK_PRESENT_MODE_FIFO_KHR.
+    For images that are presented late, and are displayed after the start of
+    the vertical blanking period (i.e. with tearing), the values of
+    sname:VkPastPresentationTimingEXT may: be treated as if the image was
+    displayed at the start of the vertical blanking period, or may: be
+    treated the same as for ename:VK_PRESENT_MODE_IMMEDIATE_KHR.
+
+Other presentation modes do not support a target presentation time or duration.
+
+.Valid Usage
+****
+  * pname:swapchainCount must: be the same value as
+    sname:VkPresentInfoKHR::pname:swapchainCount, where
+    sname:VkPresentInfoKHR is in the pname:pNext chain of this
+    sname:VkPresentTimingsInfoEXT structure.
+  * For each member of sname:VkPresentInfoKHR::pname:pSwapchains, if the
+    associated sname:VkPresentTimeEXT is non-zero, the current present mode
+    must: be ename:VK_PRESENT_MODE_FIFO_KHR or
+    ename:VK_PRESENT_MODE_FIFO_RELAXED_KHR.
+****
+
+include::{generated}/validity/structs/VkPresentTimingsInfoEXT.adoc[]
+--
+
+
+[open,refpage='VkPresentTimingInfoEXT',desc='Specifies per-present timing information',type='structs']
+--
+The sname:VkPresentTimingInfoEXT structure is defined as:
+
+include::{generated}/api/structs/VkPresentTimingInfoEXT.adoc[]
+
+  * pname:sType is a elink:VkStructureType value identifying this structure.
+  * pname:pNext is `NULL` or a pointer to a structure extending this
+    structure.
+  * pname:time is a sname:VkPresentTimeEXT specifying the target present
+    time or duration of the presentation request.
+  * pname:timeDomainId is the id of the time domain used to specify the
+    absolute target present time and the timing results obtained in a
+    subsequent flink:vkGetPastPresentationTimingEXT call for the current
+    presentation request.
+  * pname:presentStageQueries is a valid tlink:VkPresentStageFlagsEXT value
+    indicating which present stages the presentation engine should collect
+    timing information for.
+  * pname:targetPresentStage is zero or a tlink:VkPresentStageFlagsEXT value
```

Comment:
should there be language that the VkPresentStageFlagsEXT must be "valid", or one that the PE supports queries for?

Reply:
Unfortunately, no. Due to the dynamic nature of `VkSurfaceKHR` properties, implementations can't guarantee that the present stage specified at present time will be valid by the time the PE processes the presentation request. If a presentation request targets a present stage that suddenly becomes unavailable, the corresponding `VkPastPresentationRequestEXT` struct will not contain any data for the requested stage. I'll try to make that more explicit, thank you.

Reply:
This seems very unfortunate. That's exactly the type of situation VK_ERROR_OUT_OF_DATE_KHR was designed to avoid. The presentation *is* supposed to guarantee the presentation engine state doesn't change between the presentation request and actual presentation unless it returns that error.

Reply:
Seeing how this extension allows you to schedule presents really far into the future, relatively speaking, I don't know that such a guarantee can be upheld as strongly as before.